emukit/bayesian_optimization/acquisitions/expected_improvement.py

```diff
@@ -73,3 +73,69 @@ def evaluate_with_gradients(self, x: np.ndarray) -> Tuple:
     def has_gradients(self) -> bool:
         """Returns that this acquisition has gradients"""
         return isinstance(self.model, IDifferentiable)
+
+
+
+class IntegratedExpectedImprovement(Acquisition):
+
+    def __init__(self, model: Union[IModel, IDifferentiable, IPriorHyperparameters], jitter: np.float64 = np.float64(0)) -> None:
+        """
+        This acquisition computes for a given input the improvement over the current best observed value in
+        expectation. This function integrates over hyper-parameters the model by computing the  average of the
+        expected improvements for all samples. For more information see:
+
+        Efficient Global Optimization of Expensive Black-Box Functions
+        Jones, Donald R. and Schonlau, Matthias and Welch, William J.
+        Journal of Global Optimization
+
+        :param model: model that is used to compute the improvement.
+        :param jitter: parameter to encourage extra exploration.
+        """
+
+        self.model = model
+        self.jitter = jitter
+
```

Comment:
It is better to make samples a part of acquisition function state, like so:

```suggestion
self.samples = self.model.generate...
```

and consequently remove them from the model